File: front/code-book/src/types/fixture/book.ts
import { Book } from '../book'
import { fileCode, snippetCode, urlCode } from './code'
import { scala } from './tag'
import { cat } from './user'

export const fullBook: Book = {
  id: '11111111-1111-4111-8111-111111111111',
  emoji: '🖼',
  title: 'フロントコード集',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  author: cat,
  tags: [scala],
  created: '2022/09/01 12:34:56',
  codes: [fileCode, urlCode, snippetCode],
}

File: back/code-book/src/main/java/com/codebook/codebook/book/Book.java
package com.codebook.codebook.book;

import com.codebook.codebook.code.Code;
import com.codebook.codebook.user.User;
import lombok.Value;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Value
public class Book {

    String id;

    String emoji;

    String title;

    String description;

    User author;

    String created;

    List<Code> codes;

    public Map<String, Object> toMap() {
        return Map.of(
            "book", Map.of(
                "id", id,
                "emoji", emoji,
                "title", title,
                "description", description,
                "author", author.toMap(),
                "tags", codes.stream().flatMap(c -> c.getPart().getTags().stream()).distinct().collect(Collectors.toList()),
                "created", created,
                "codes", codes.stream().map(Code::toMap).collect(Collectors.toList())
            )
        );
    }

}
